src/libLookup/Lookup.cpp

```diff
@@ -2455,6 +2455,34 @@ void Lookup::StartSynchronization() {
   DetachedFunction(1, func);
 }
 
+bool Lookup::FetchDSInfoLoop() {
+  unsigned int counter = 0;
+  {
+    lock_guard<mutex> g(m_mediator.m_mutexDSCommittee);
+    if (m_mediator.m_DSCommittee->size() > 0) {
+      LOG_GENERAL(WARNING,
+                  "DS comm already set, make sure you cleaned variables");
+      return false;
+    }
+  }
+
+  while (counter <= FETCH_LOOKUP_MSG_MAX_RETRY) {
+    GetDSInfoFromLookupNodes();
+    this_thread::sleep_for(chrono::seconds(NEW_NODE_SYNC_INTERVAL));
+    {
+      lock_guard<mutex> g(m_mediator.m_mutexDSCommittee);
+      if (m_mediator.m_DSCommittee->size() > 0) {
+        return true;
+      }
+    }
+    counter++;
+  }
```

Comment:
How about using cv instead of polling asking? It should be more efficient. You can refer to this: https://github.com/Zilliqa/Zilliqa/blob/master/src/libNode/Node.cpp#L316